orangecontrib/text/sentiment/__init__.py

```diff
@@ -0,0 +1,71 @@
+import scipy.sparse as sp
+
+from nltk.corpus import opinion_lexicon
+from nltk.sentiment import SentimentIntensityAnalyzer
+from orangecontrib.text import Corpus
+from orangecontrib.text.preprocess import WordPunctTokenizer
+from orangecontrib.text.vectorization.bagofwords import BoWPreprocessTransform, BoWComputeValue
+from orangecontrib.text.vectorization.base import BaseVectorizer
+
+
+class Liu_Hu_Sentiment(BaseVectorizer):
+    positive = set(opinion_lexicon.positive())
+    negative = set(opinion_lexicon.negative())
+    sentiments = ('sentiment')
+
+    def __init__(self):
+        super().__init__()
+        self.dic = {0: 'sentiment'}
+        self.tokens = None
+
+    def _transform(self, corpus, source_dict=None):
+        scores = []
+        if self.tokens is None:
+            self.set_tokens(corpus)
+        for doc in self.tokens:
+            pos_words = sum(word in self.positive for word in doc)
+            neg_words = sum(word in self.negative for word in doc)
+            scores.append([100*(pos_words - neg_words)/max(len(doc), 1)])
+        X = sp.csr_matrix(scores)
+
+        # set  compute values
+        shared_cv = BoWPreprocessTransform(corpus.used_preprocessor, self, self.dic)
+        cv = [BoWComputeValue(self.dic[i], shared_cv) for i in range(len(self.dic))]
+
+        self.add_features(corpus, X, self.dic, cv)
+        return corpus
+
+    def set_tokens(self, corpus):
+        tokenizer = WordPunctTokenizer()
+        self.tokens = tokenizer(corpus.documents)
+
+
+class Vader_Sentiment(BaseVectorizer):
+    sentiments = ('pos', 'neg', 'neu', 'compound')
+
+    def __init__(self):
+        super().__init__()
+        self.vader = SentimentIntensityAnalyzer()
+        self.dic = dict(enumerate(self.sentiments))
+
+    def _transform(self, corpus, source_dict=None):
+        scores = []
+        for doc in corpus:
+            text = '\n'.join([doc[tf].value for tf in corpus.text_features])
+            pol_sc = self.vader.polarity_scores(text)
+            scores.append([pol_sc[x] for x in self.sentiments])
+        X = sp.csr_matrix(scores)
+
+        # set  compute values
+        shared_cv = BoWPreprocessTransform(corpus.used_preprocessor, self, self.dic)
```

Comment:
If I got this right, `corpus.used_preprocessor` isn't really used. If so, rather you `lambda x: x` or something more obvious.